fritzconnection/cli/fritzwol.py

```diff
@@ -0,0 +1,73 @@
+"""
+fritzwol.py
+
+Module to wake up a single host via the Fritzbox built-in mechanism.
+This can be helpful if the host to be woken up is in a different
+broadcast domain/ subnet than the client which tries to wake up.
+CLI interface.
+
+This module is part of the FritzConnection package.
+https://github.com/kbr/fritzconnection
+License: MIT (https://opensource.org/licenses/MIT)
+Author: Maik Töpfer
+"""
+
+from fritzconnection.core.exceptions import FritzConnectionException
+from fritzconnection.core.fritzconnection import FritzConnection
+from ..lib.fritzhosts import FritzHosts
+from . utils import get_cli_arguments, get_instance, print_header
+
+
+class WakeOnLan:
+    def __init__(self, fc, fh):
+        self._fc = fc
+        self._fh = fh
+
+    def wakeup(self, host_name):
+        print(f"Waking up host '{host_name}'...")
+        mac = self._get_mac_address(host_name)
+        self._wakeup_host(mac)
+        print("Done")
+
+    def _get_mac_address(self, host_name):
+        mac = [host['mac']
```

Comment:
You are iterating over all devices, building a list and just use the first match and discarding all others. It would be better to rewrite 

```Python
for host in self._fh.get_hosts_info():
    if host["name"] == host_name:
        return host["mac"]
raise WakeOnLanException(...)
```